src/pymor/operators/block.py

```diff
@@ -245,3 +253,243 @@ def assemble_lincomb(self, operators, coefficients, solver_options=None, name=No
+        else:
+            c = coefficients[0]
+            if c == 1:
+                return self
+            for (i, j) in np.ndindex(self._blocks.shape):
+                blocks[i, j] = self._blocks[i, j] * c
+            return BlockOperator(blocks)
+
+
+class ShiftedSecondOrderSystemOperator(BlockOperator):
+    """BlockOperator appearing in second-order systems.
+
+    This represents a block operator
+
+    .. math::
+        a \mathcal{E} + b \mathcal{A} =
+        \begin{bmatrix}
+            a I & b I \\
+            -b K & a M - b D
+        \end{bmatrix},
+
+    which satisfies
+
+    .. math::
+        (a \mathcal{E} + b \mathcal{A})^T
+        &=
+        \begin{bmatrix}
+            a I & -b K^T \\
+            b I & a M^T - b D^T
+        \end{bmatrix}, \\
+        (a \mathcal{E} + b \mathcal{A})^{-1}
+        &=
+        \begin{bmatrix}
+            (a^2 M - a b D + b^2 K)^{-1} (a M - b D)
+            & -b (a^2 M - a b D + b^2 K)^{-1} \\
+            b (a^2 M - a b D + b^2 K)^{-1} K
+            & a (a^2 M - a b D + b^2 K)^{-1}
+        \end{bmatrix}, \\
+        (a \mathcal{E} + b \mathcal{A})^{-T}
+        &=
+        \begin{bmatrix}
+            (a M - b D)^T (a^2 M - a b D + b^2 K)^{-T}
+            & b K^T (a^2 M - a b D + b^2 K)^{-T} \\
+            -b (a^2 M - a b D + b^2 K)^{-T} & a (a^2 M - a b D + b^2 K)^{-T}
+        \end{bmatrix}.
+
+    Parameters
+    ----------
+    M
+        |Operator|.
+    D
+        |Operator|.
+    K
+        |Operator|.
+    p
+        Complex number.
+    """
+
+    def __init__(self, M, D, K, a, b):
+        super().__init__([[IdentityOperator(M.source) * a, IdentityOperator(M.source) * b],
+                          [K * (-b), LincombOperator([M, D], [a, -b])]])
+        self.M = M
+        self.D = D
+        self.K = K
+        self.a = a
+        self.b = b
+
+    def apply(self, U, mu=None):
+        assert U in self.source
+        V_blocks = [U.block(0) * self.a + U.block(1) * self.b,
+                    self.K.apply(U.block(0), mu=mu) * (-self.b) +
+                    self.M.apply(U.block(1), mu=mu) * self.a -
+                    self.D.apply(U.block(1), mu=mu) * self.b]
+        return self.range.make_array(V_blocks)
+
+    def apply_transpose(self, V, mu=None):
+        assert V in self.range
+        U_blocks = [V.block(0) * self.a - self.K.apply_transpose(V.block(1), mu=mu) * self.b,
+                    V.block(0) * self.b +
+                    self.M.apply_transpose(V.block(1), mu=mu) * self.a -
+                    self.D.apply_transpose(V.block(1), mu=mu) * self.b]
+        return self.source.make_array(U_blocks)
+
+    def apply_inverse(self, V, mu=None, least_squares=False):
+        assert V in self.range
+        aMmbDV0 = self.M.apply(V.block(0), mu=mu) * self.a - self.D.apply(V.block(0), mu=mu) * self.b
+        KV0 = self.K.apply(V.block(0), mu=mu)
+        a2MmabDpb2K = LincombOperator([self.M, self.D, self.K], [self.a ** 2, -self.a * self.b, self.b ** 2])
```

Comment:
You might want to call `assemble` on the `LincombOperator`. Otherwise, it is assembled multiple times ..

Reply:
Yes, good point.